pandas/tests/scalar/timedelta/test_arithmetic.py

```diff
@@ -200,6 +200,48 @@ def test_td_rsub_numeric_raises(self):
         with pytest.raises(TypeError):
             2.0 - td
 
+    def test_td_sub_timedeltalike_object_dtype_array(self):
+        # GH 21980
+        arr = np.array([Timestamp('20130101 9:01'),
+                        Timestamp('20121230 9:02')])
+        exp = np.array([Timestamp('20121231 9:01'),
+                        Timestamp('20121229 9:02')])
+        res = arr - pd.Timedelta('1D')
+        tm.assert_numpy_array_equal(res, exp)
+
+    def test_td_sub_mixed_most_timedeltalike_object_dtype_array(self):
+        # GH 21980
+        now = pd.Timestamp.now()
+        arr = np.array([now,
+                        pd.Timedelta('1D'),
+                        np.timedelta64(2, 'h')])
+        exp = np.array([now - pd.Timedelta('1D'),
+                        pd.Timedelta('0D'),
+                        np.timedelta64(2, 'h') - pd.Timedelta('1D')])
+        res = arr - pd.Timedelta('1D')
```

Comment:
can you also tests the rsub (flip the args)

Reply:
i think Timedelta - Timestamp throws an exception iirc

Reply:
which if that is true i think makes sense? unless td - timestamp should be some value

Reply:
I'm guessing jreback's suggestion is to test exactly that, that it raises a TypeError

Reply:
yes - you also should test ops and their reverse

Reply:
done